### ISSUE TITLE
`organizeDeclarations` misplaces instance function in the Lifecycle category

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#728]
The `organizeDeclarations` is misplacing function declarations with nested types in the `Lifecycle` category:

```diff
struct Foo {

-    // MARK: Internal
+    // MARK: Lifecycle

    func bar() {
        class NestedClass {
             // `organizeDeclarations` sees this `init` token in the body of `Foo.bar()`, 
             // which causes it to place this function in the Lifecycle category (where an initializer would go)
             init() { }
        }

       // ...
    }

}

```

**Comment by calda** [comment]
I'll post a fix for this when I have a chance.

**Comment by nicklockwood** [comment]
@calda fixed in 0.46.2.

**Comment by calda** [comment]
Thank you!
